lib/tests/streamlit/web/server/routes_test.py

```diff
@@ -172,3 +180,26 @@ def test_allowed_message_origins(self):
         self.assertEqual(
             {"allowedOrigins": ALLOWED_MESSAGE_ORIGINS}, json.loads(response.body)
         )
+
+    # NOTE: Temporary tests to verify this endpoint can also act as a healthcheck
+    # endpoint while we need it to. These tests are more or less copy-paste from the
+    # HealthHandlerTest class above.
+    def test_healthcheck_responsibilities(self):
+        response = self.fetch("/st-allowed-message-origins")
+        self.assertEqual(200, response.code)
+
+        self._is_healthy = False
+        response = self.fetch("/st-allowed-message-origins")
+        self.assertEqual(503, response.code)
+
+    def test_healthcheck_responsibilities_without_csrf(self):
+        config._set_option("server.enableXsrfProtection", False, "test")
```

Comment:
Nit: use `with patch_config_options({"server.enableXsrfProtection": False}):` instead of `config._set_option` to avoid "permanently" mutating the config option for any test executing later in the test run